### PR TITLE
feat: add `push-to-huggingface` command

### LINKED ISSUE #3670: [FEATURE] Add push to Hugging Face hub command `python -m argilla datasets push-to-hf`

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

This PR adds the `datasets push-to-huggingface` command to the Argilla CLI, to push a `FeedbackDataset` from Argilla into the HuggingFace Hub.

Closes #3670 

**Type of change**

- [X] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Add unit tests for the `datasets push-to-huggingface` command

**Checklist**

- [ ] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/tasks/datasets/push.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    try:
+        console.print(
+            f":two: Pushing `FeedbackDataset` with name={name} to the HuggingFace Hub..."
+            if not workspace
+            else f":two: Pushing `FeedbackDataset` with name={name} and workspace={workspace} to the HuggingFace Hub..."
+        )
+        dataset.push_to_huggingface(repo_id=repo_id, private=private, token=token)
+        console.print(
+            f":sparkles: `FeedbackDataset` successfully pushed to the HuggingFace Hub at https://huggingface.co/{repo_id}"
```
Comment (gabrielmbmb) [inline src/argilla/tasks/datasets/push.py]:
```suggestion
            f":sparkles: `FeedbackDataset` successfully pushed to the :hugs: HuggingFace Hub at https://huggingface.co/{repo_id}"
```

File: src/argilla/tasks/datasets/push.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            " sure to add records before pushing it."
+        )
+        raise typer.Exit(1) from e
+    except Exception as e:
+        typer.echo("An unexpected error occurred when trying to push the `FeedbackDataset` to the HuggingFace Hub")
+        raise typer.Exit(code=1) from e
+
+
+if __name__ == "__main__":
+    typer.run(push_to_hf)
```
Comment (gabrielmbmb) [inline src/argilla/tasks/datasets/push.py]: we can remove this

File: tests/unit/tasks/datasets/conftest.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+def remote_feedback_dataset() -> "RemoteFeedbackDataset":
+    workspace = Workspace.__new__(Workspace)
+    workspace.__dict__.update(
+        {
+            "id": uuid4(),
+            "name": "unit-test",
+            "inserted_at": datetime.now(),
+            "updated_at": datetime.now(),
+        }
+    )
```
Comment (gabrielmbmb) [inline tests/unit/tasks/datasets/conftest.py]: I've added one `workspace` fixture in https://github.com/argilla-io/argilla/pull/3688, maybe we can merge that PR first and you can reuse the fixture here

File: tests/unit/tasks/datasets/test_list.py [hunk 4]
```diff
@@ -12,69 +12,30 @@
...
-        owner="unit-test",
-        workspace="unit-test",
-        created_at=datetime.now(),
-        last_updated=datetime.now(),
-    )
-
-
 @pytest.mark.usefixtures("login_mock")
 class TestSuiteListDatasetsCommand:
+    @pytest.mark.usefixtures("remote_feedback_dataset", "dataset")
```
Comment (gabrielmbmb) [inline tests/unit/tasks/datasets/test_list.py]:
```suggestion
```
no need to have this

File: tests/unit/tasks/datasets/test_push.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+    from click.testing import CliRunner
+    from pytest_mock import MockerFixture
+    from typer import Typer
+
+
+@pytest.mark.usefixtures("login_mock")
+class TestSuiteDatasetsPushCommand:
+    @pytest.mark.usefixtures("remote_feedback_dataset")
```
Comment (gabrielmbmb) [inline tests/unit/tasks/datasets/test_push.py]:
```suggestion
```

**Review by gabrielmbmb — APPROVED** [review]
LGTM! Just some comments about the fixtures and adding the :hugs: emoji

File: src/argilla/tasks/datasets/push.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
```
Comment (alvarobartt) [inline src/argilla/tasks/datasets/push.py]:
```suggestion
```

File: tests/unit/tasks/datasets/conftest.py (hunk 3 above)
Comment (alvarobartt) [reply tests/unit/tasks/datasets/conftest.py]: Sure, we can wait to merge this PR 👍🏻

**Comment by tomaarsen** [comment]
Don't we use `huggingface` instead of `hf` elsewhere in the project?

File: CHANGELOG.md [hunk 7]
```diff
@@ -35,6 +35,7 @@ These are the section headers that we use:
 - Added `users create` command to create users ([#3667](https://github.com/argilla-io/argilla/pull/3667)).
 - Added `whoami` command to get current user ([#3673](https://github.com/argilla-io/argilla/pull/3673)).
 - Added `users delete` command to delete users ([#3671](https://github.com/argilla-io/argilla/pull/3671)).
+- Added `datasets push-to-hub` command to push a `FeedbackDataset` from Argilla into the HuggingFace Hub ([#3685](https://github.com/argilla-io/argilla/pull/3685)).
```
Comment (tomaarsen) [inline CHANGELOG.md]:
```suggestion
- Added `datasets push-to-hf` command to push a `FeedbackDataset` from Argilla into the HuggingFace Hub ([#3685](https://github.com/argilla-io/argilla/pull/3685)).
```
This is certainly wrong

**Comment by alvarobartt** [comment]
> Don't we use `huggingface` instead of `hf` elsewhere in the project?

True, but for the command `push-to-huggingface` seems a bit long, anyway we can discuss this to see what's better between `push-to-hub`, `push-to-hf` or `push-to-huggingface`

File: CHANGELOG.md (hunk 7 above)
Comment (alvarobartt) [reply CHANGELOG.md]: thanks! solved it as part of the merge from `develop`, as there were some conflicts in the `CHANGELOG.md`

File: tests/unit/tasks/datasets/conftest.py (hunk 3 above)
Comment (gabrielmbmb) [reply tests/unit/tasks/datasets/conftest.py]: it has been merged, do you want to update it?

**Pull request merged**